hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/reader/record/RecordFileReaderImplV5.java

```diff
@@ -107,19 +108,39 @@ private void readBody(ValidatedDataInputStream vdis, DigestInputStream metadataD
         long consensusStart = 0;
         RecordStreamObject lastRecordStreamObject = null;
         List<RecordItem> items = new ArrayList<>();
+        RecordItem lastRecordItem = null;
 
         // read record stream objects
         while (!isHashObject(vdis, hashObjectClassId)) {
             RecordStreamObject recordStreamObject = new RecordStreamObject(vdis, recordFile.getHapiVersion(),
                     count);
-            items.add(recordStreamObject.getRecordItem());
+            var recordItem = recordStreamObject.getRecordItem();
+
+            // check if current item is a child
+            if (recordItem.isChild()) {
+                if (lastRecordItem != null &&
+                        DomainUtils.timestampInNanosMax(recordItem.getRecord().getParentConsensusTimestamp())
+                                .longValue() ==
+                                DomainUtils.timestampInNanosMax(lastRecordItem.getRecord().getConsensusTimestamp())
+                                        .longValue()) {
```

Comment:
Can be simplified and avoid the unnecessary timestamp conversion:

```suggestion
            if (recordItem.isChild() && parentRecordItem != null && recordItem.getParentConsensusTimestamp().equals(parentRecordItem.getConsensusTimestamp()) {
```

Reply:
Updated